OpenWeather/Data/Adapters/LocationAdapter.swift

```diff
@@ -7,12 +7,42 @@
 
 import Foundation
 import CoreLocation
+import GRDB
+
+enum LocationAdapter {
+    static func device(from: LocationData) -> DeviceLocation {
+        DeviceLocation(
+            name: from.name,
+            latitude: from.lat,
+            longitude: from.long
+        )
+    }
+
+    static func device(from: Row) throws -> DeviceLocation {
+        guard let name = from[0] as? String else {
+            throw DatabaseError.noFoundColumn("name")
+        }
+
+        guard let lat = from[1] as? Double else {
+            throw DatabaseError.noFoundColumn("lat")
+        }
+
+        guard let long = from[2] as? Double else {
+            throw DatabaseError.noFoundColumn("long")
+        }
```

Comment:
Awesome! This is great. Could we add a guard at the beginning to check that the Row contains at least 3 items?